src/database/Database.cpp

```diff
@@ -119,6 +119,7 @@ Database::applySchemaUpgrade(unsigned long vers)
 
     case 4:
         BanManager::dropAll(*this);
+        DataFrame::dropAll(*this); // not and error, it changed again
```

Comment:
lastmodified is used for a couple things:
* meta data generation
* history traversal (allows to skip to the right ledger to build the history for that ledger entry)

the problem with the drop table is more that we "leak" entries in the bucket list, and with that potentially generate the wrong deltas (which will cause the ledger hashes to differ between machines and therefore invalidate the ledger chain)